pkg/snapstore/abs_snapstore.go

```diff
@@ -347,30 +349,60 @@ func (a *ABSSnapStore) Delete(snap brtypes.Snapshot) error {
 	return nil
 }
 
-// ABSSnapStoreHash calculates and returns the hash of azure object storage snapstore secret.
-func ABSSnapStoreHash(config *brtypes.SnapstoreConfig) (string, error) {
+// ABSSnapStoreModTime returns the last modification time of the credential files
+func ABSSnapStoreModTime(config *brtypes.SnapstoreConfig) (time.Time, error) {
 	if dir, isSet := os.LookupEnv(absCredentialFile); isSet {
-		absConfig, err := readABSCredentialFiles(dir)
+		absTimeStamp, err := readABSCredentialModTime(dir)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials from %v directory", dir)
+			return time.Time{}, fmt.Errorf("failed to get ABS credentials from the directory %v with error: %v", dir, err.Error())
 		}
-		return getABSHash(absConfig), nil
+		return absTimeStamp, nil
 	}
 
 	if filename, isSet := os.LookupEnv(absCredentialJSONFile); isSet {
-		absConfig, err := readABSCredentialsJSON(filename)
+		fileInfo, err := os.Stat(filename)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials using %v file", filename)
+			return time.Time{}, fmt.Errorf("failed to fetch file information of the ABS JSON credential file %v with error: %v", filename, err.Error())
 		}
-		return getABSHash(absConfig), nil
+		absTimeStamp, err := readModTime(fileInfo)
+		if err != nil {
+			return time.Time{}, fmt.Errorf("failed to read file information of the ABS JSON credential file %v with error: %v", filename, err.Error())
+		}
+		return absTimeStamp, nil
 	}
 
-	return "", nil
+	return time.Time{}, fmt.Errorf("No environment variable set for the ABS credential file")
```

Comment:
The behavior now changes since this function returns error when the env var is not present, rather than returning nil as it used to do previously, please add some unit tests to test the behavior after this change. Same applies to all providers.